src/shared/sol-util.c

```diff
@@ -905,3 +905,79 @@ sol_util_uuid_str_valid(const char *str)
 
     return true;
 }
+
+SOL_API int
+sol_util_escape_quotes(const struct sol_str_slice slice,
```

Comment:
I think we should change the return to `ssize_t` and return the index in slice to the last used char, so we can use it to process the information further. This is particularly of help if we use the `x"y"` to only get `x`, then user would get the pointer to the first quote and may be able to handle it.